README.md

```diff
@@ -1,32 +1,574 @@
+      { authToken: data.token },
+      function(err, data) { console.log(data.profile.basic_info.full_name) }
+    )
+})
+```
+
+and another one?
+
+```
+ajax('https://auth.uniplaces.com', function(err, authData) {
+    ajax(
+        'https://core.uniplaces.com/api/guests/cc208a7a-cea6-4574-9fed-95ee27060689',
+        function(err, guestData) {
+            ajax(
+              'https://core.uniplaces.com/api/pay',
+              function(err, paymentData) {
+                redirectToSuccessPage()
+              }
+            )
+        }
+    )
+})
+```
+
+### callback hell
+
+Callbacks come with some problems, one of them being loosing control of the
+invocation of the callback. It's whatever tool, library, wtv you're using that
+is calling the callback, so you just cross fingers and hope everything comes out
+fine.
+
+The other is lack os trustability, readability and sequentiability.
+
+## Promises
+
+What are promises? Promises are future values. They give you an API so you can
+know when a task finishes so you can decide what happens afterwards. Most of you
+have used it, our JS projects use it a lot. New APIs to the browser and dom are
+being built on Promises, it is critical for a JS developer nowadays to know
+about them. But why are promises so cool?
+
+They solve most of the problems created by callbacks: Inversion of control,
+instead of being _someone_ calling your function, it's the developer saying "do
+this `then` that", it is you executing the callback.
+
+### Yeah yeah, I've used Promises, I know about that, what are you teaching me?
+
+Promises are one of those things that people tend to use without really
+understanding (javascript being other of those things), and that happens because
+their API is so painfully obvious that is just **makes sense**.
+
+But let's try to go a little bit deeper in the abstraction
+
+### The future value
+
+Imagine we go to McDonalds, you choose your cheeseburger, and you get a paper
+with a number on it. That paper grants you that you will get your cheeseburger,
+that paper is a _future cheeseburger_. When the cashier calls your number, you
+can exchange your _promise of a cheesburguer_ by the cheesburger itself.
+
+## Let's come back to our ajax example
+
+```
+ajax('https://auth.uniplaces.com', function(err, authData) {
+    ajax(
+        'https://core.uniplaces.com/api/guests/cc208a7a-cea6-4574-9fed-95ee27060689',
+        function(err, guestData) {
+            ajax(
+              'https://core.uniplaces.com/api/pay',
+              function(err, paymentData) {
+                redirectToSuccessPage()
+              }
+            )
+        }
+    )
+})
+```
+
+And write it using promises
+
+```
+promisifiedAjax('https://auth.uniplaces.com')
+  .then(promisifiedAjax('https://core.uniplaces.com/api/guests/cc208a7a-cea6-4574-9fed-95ee27060689'))
+  .then(promisifiedAjax('https://core.uniplaces.com/api/pay'))
+  .then(redirectToSuccess)
```

Comment:
Function calls are wrong

Reply:
Wrong as in should not be used for sake code readability ?

Reply:
Not only by that but because calling them as I'm calling them here would actually execute this at the *declare time*. It should be `.then(() => promisifiedAjax('https://core.uniplaces.com/api/pay'))`